site/en/posts/site-refactor-2022.md

```diff
@@ -0,0 +1,92 @@
+
+After some investigation, we wound up choosing to [migrate to an image CDN](https://github.com/chromeos/chromeos.dev/issues/154). This did a number of things for us:
+
+- Removed the need for us to store images in our codebase,
+- Removed image manipulation from our build stack.
+- Removed the need for us to use the `picture` tag for all of our images, making our CSS and HTML easier to maintain.
+
+It did all of this while resolving all of our requirements and (mostly) without changing any of the HTML or markdown already written for the site. Our whole image optimization pipeline, reduced to essentially the same PostHTML transform we had already written for Vite Plugin Img. There was only one change required: images that use our CDN need to be prefixed with `ix://` for the transform to be applied. A small inconvenience; we already had precedent for image prefixes in our codebase and the benefits were clear.
+
+## Other Updates
+
+There were a few additional updates we wound up doing based on contributor feedback.
+
+First, we removed commit linting all together; it didn't work reliably across all operating systems and added a lot of developer overhead. Instead, we implemented PR title checking and PR squashing, letting all commits roll up into a single, tidy linted message. We even [kept emojis and optional scopes](https://github.com/chromeos/chromeos.dev/blob/77c717c3b752be7276f4d4b0185da9f792377c54/.github/pr-title-checker-config.json).
+
+Next, we changed our linting git hook. Previously, we ran linting over all files before a commit could happen. Now, we only lint the code that's staged, and rely on our continuous integration tests to catch anything that may slip through the cracks.
+
+Finally, we [overhauled our service worker](https://github.com/chromeos/chromeos.dev/issues/159). We implemented [broadcast updates](https://developers.google.com/web/tools/workbox/modules/workbox-broadcast-update) to let users know if the cached content they're viewing is out-of-date, we made the home page and the news and stories landing pages try the network before falling back to cache, and we changed how our assets get cached during installation and runtime. Because our images not are from a CDN, we also needed to implement [opaque response caching](https://developers.google.com/web/tools/workbox/guides/handle-third-party-requests#workbox_caches_opaque_response_sometimes) for them.
```

Comment:
```suggestion
Finally, we [overhauled our service worker](https://github.com/chromeos/chromeos.dev/issues/159). We implemented [broadcast updates](https://developers.google.com/web/tools/workbox/modules/workbox-broadcast-update) to let users know if the cached content they're viewing is out-of-date, we made the home page and the news and stories landing pages try the network before falling back to cache, and we changed how our assets get cached during installation and runtime. Because our images are not from a CDN, we also needed to implement [opaque response caching](https://developers.google.com/web/tools/workbox/guides/handle-third-party-requests#workbox_caches_opaque_response_sometimes) for them.
```